enlighted-link.html

```diff
@@ -1,28 +1,39 @@
 <!-- enlighted-link @version: 0.0.0 @license: MIT -->
 <script>
-    customElements.define('enlighted-link', class extends HTMLElement {
-        // List of attributes observed for attributeChangedCallback
-        static get observedAttributes() {
-            return [];
-        }
+customElements.define('enlighted-link', class extends HTMLElement{
+    static get observedAttributes(){
+        return [
+            // content attrs
+            'href',
+            'rel',
+            'media',
+            'hreflang',
+            'type',
+            'sizes',
+            'title'
+        ]
+    }
+    connectedCallback(){
+        const link = this.link = this.link || document.createElement('link');
 
-        // Fires when an instance of the element is created
-        constructor() {
-            super();
+        for(let attrNo = 0, len = this.attributes.length; attrNo < len; attrNo++){
+            link.setAttribute(this.attributes[attrNo].name, this.attributes[attrNo].value);
         }
-        // Fires when an instance was inserted into the document
-        connectedCallback() {
-
+        document.head.appendChild(link);
+        // TODO/FIXME(tomalec): attach mutation observer on links parent, to detach this in case link is detached
```

Comment:
I don't think you need a Mutation Observer. Just do

`this.link.addEventListener('DOMNodeRemoved', this.remove());`

See: https://jsfiddle.net/vqrfewfh/

Reply:
`MutationEvents` are deprecated in favor of `MutationObserver`. I'd not add feature based on an already deprecated thing. Given MO could be expensive, I'd stick to YAGNI, and just not implement this feature for now. Therefore change `TODO/FIXME` to `IDEA`. WDYT?

Reply:
Good points. And now I read 

"Adding DOM mutation listeners to a document profoundly degrades the performance of further DOM modifications to that document (making them 1.5 - 7 times slower!). Moreover, removing the listeners does not reverse the damage."

from: https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Mutation_events

This is surprising to me!